src/mlpack/bindings/julia/mlpack/cli.jl

```diff
@@ -0,0 +1,364 @@
+    input
+  else
+    convert_to_1d(convert(Array{T, 2}, input))
+  end
+end
+
+function CLIRestoreSettings(programName::String)
+  ccall((:CLI_RestoreSettings, library), Nothing, (Cstring,), programName);
+end
+
+function CLISetParam(paramName::String, paramValue::Int)
+  ccall((:CLI_SetParamInt, library), Nothing, (Cstring, Int), paramName,
+      paramValue);
+end
+
+function CLISetParam(paramName::String, paramValue::Float64)
+  ccall((:CLI_SetParamDouble, library), Nothing, (Cstring, Float64), paramName,
+      paramValue);
+end
+
+function CLISetParam(paramName::String, paramValue::Bool)
+  ccall((:CLI_SetParamBool, library), Nothing, (Cstring, Bool), paramName,
+      paramValue);
+end
+
+function CLISetParam(paramName::String, paramValue::String)
+  ccall((:CLI_SetParamString, library), Nothing, (Cstring, Cstring), paramName,
+      paramValue);
+end
+
+function CLISetParamMat(paramName::String,
+                        paramValue,
+                        pointsAsRows::Bool)
+  paramMat = to_matrix(paramValue, Float64)
+  ccall((:CLI_SetParamMat, library), Nothing, (Cstring, Ptr{Float64}, UInt64,
+      UInt64, Bool), paramName, Base.pointer(paramMat), size(paramMat, 1),
+      size(paramMat, 2), pointsAsRows);
+end
+
+function CLISetParamUMat(paramName::String,
+                         paramValue,
+                         pointsAsRows::Bool)
+  paramMat = to_matrix(paramValue, Int64)
+
+  # Sanity check.
+  if minimum(paramMat) <= 0
+    throw(DomainError("Input $(paramName) cannot have 0 or negative values!  " *
+        "Must be 1 or greater."))
+  end
+
+  m = convert(Array{UInt64, 2}, paramMat .- 1)
+  ccall((:CLI_SetParamUMat, library), Nothing, (Cstring, Ptr{UInt64}, UInt64,
+      UInt64, Bool), paramName, Base.pointer(m), size(paramValue, 1),
+      size(paramValue, 2), pointsAsRows);
+end
+
+function CLISetParam(paramName::String,
+                     vector::Vector{String})
+  # For this we have to set the size of the vector then each string
+  # sequentially.  I am not sure if this is fully necessary but I have some
+  # reservations about Julia's support for passing arrays of strings correctly
+  # as a const char**.
+  ccall((:CLI_SetParamVectorStrLen, library), Nothing, (Cstring, UInt64),
+      paramName, size(vector, 1));
+  for i in 1:size(vector, 1)
+    ccall((:CLI_SetParamVectorStrStr, library), Nothing, (Cstring, Cstring,
+        UInt64), paramName, vector[i], i .- 1);
+  end
+end
+
+function CLISetParam(paramName::String,
+                     vector::Vector{Int64})
+  ccall((:CLI_SetParamVectorInt, library), Nothing, (Cstring, Ptr{Int64},
+      Int64), paramName, Base.pointer(vector), size(vector, 1));
+end
+
+function CLISetParam(paramName::String,
+                     matWithInfo::Tuple{Array{Bool, 1}, Array{Float64, 2}},
+                     pointsAsRows::Bool)
+  ccall((:CLI_SetParamMatWithInfo, library), Nothing, (Cstring, Ptr{Bool},
+      Ptr{Float64}, Int64, Int64, Bool), paramName,
```

Comment:
I am not familiar we Julia but can we add a sanity check for dimensions of bool array must be equal to float array or Julia will itself check this.?